x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/InstanceTypeService.java

```diff
@@ -20,40 +22,46 @@
 import java.util.zip.GZIPInputStream;
 
 public final class InstanceTypeService {
-
     private InstanceTypeService() {}
 
     private static final class Holder {
         private static final Map<InstanceType, CostEntry> costsPerDatacenter;
-
         static {
+            final Logger log = LogManager.getLogger(TransportGetStackTracesAction.class);
```

Comment:
This should use `InstanceTypeService.class` instead of `TransportGetStackTracesAction.class`. Also, is there a reason why you declared `log` just here in the static block? It's usual to declare this as a class constant (even though you only use it here). I'd also just pull the logger up to the enclosing class `InstanceTypeService`.

Reply:
:+1: Class changed.

Re scope of logger: If I understood correctly, this static code is only run once. Thus, the logger is only called once and should be GC'ed after use.

Reply:
On the technical aspects you're correct but it's rather unusual to do that. It's not that the logger will add any noticeable overhead if we declare it as class constant on top level. Furthermore, if we change this class in the future we might want to log in other places as well and then we need to remember to pull this logger to the top level.

Reply:
I'll change it to unblock the PR for 8.13, even when I have a different opinion here =)

To be consistent with naming of constants, shouldn't the variable name should from `log` to `LOG`? Is lowercase `log` a generally accepted exception? Do we have more of these exceptions in the ES code base?

Reply:
I'd say `log` is the only exception to that rule that I'm aware of.